platform/core/src/services/CustomizationService/CustomizationService.ts

```diff
@@ -235,292 +173,333 @@ export default class CustomizationService extends PubSubService {
+  ): void {
+    // if (typeof customization === 'string') {
+    //   const extensionValue = this._findExtensionValue(customization);
+    //   customization = extensionValue.value;
+    // }
+
+    switch (scope) {
+      case CustomizationScope.Global:
+        this.setGlobalCustomization(customizationId, customization);
+        break;
+      case CustomizationScope.Mode:
+        this.setModeCustomization(customizationId, customization);
+        break;
+      case CustomizationScope.Default:
+        this.setDefaultCustomization(customizationId, customization);
+        break;
+      default:
+        throw new Error(`Invalid customization scope: ${scope}`);
+    }
+  }
+
+  /**
+   * Gets all customizations for a given scope.
    *
-   * @param defaultValue to return if no customization specified.
+   * @param scope - The scope to retrieve customizations from: 'global', 'mode', or 'default'
+   * @returns A Map containing all customizations for the specified scope
    */
-  public getModeCustomization = this.getCustomization;
+  public getCustomizations(scope: CustomizationScope): Map<string, Customization> {
+    if (scope === CustomizationScope.Global) {
+      return this.globalCustomizations;
+    }
+    if (scope === CustomizationScope.Mode) {
+      return this.modeCustomizations;
+    }
+    return this.defaultCustomizations;
+  }
 
   /**
    *  Returns true if there is a mode customization.  Doesn't include defaults, but
    * does return global overrides.
    */
-  public hasModeCustomization(customizationId: string) {
+  public hasCustomization(customizationId: string) {
     return (
       this.globalCustomizations.has(customizationId) || this.modeCustomizations.has(customizationId)
     );
   }
 
-  /**
-   * get is an alias for getModeCustomization, as it is the generic getter
-   * which will return both mode and global customizations, and should be
-   * used generally.
-   * Note that the second parameter, defaultValue, will be expanded to include
-   * any customizationType values defined in it, so it is not the same as doing:
-   *   `customizationService.get('key') || defaultValue`
-   * unless the defaultValue does not contain any customizationType definitions.
-   */
-  public get = this.getModeCustomization;
-
   /**
    * Applies any inheritance due to UI Type customization.
-   * This will look for customizationType in the customization object
+   * This will look for inheritsFrom in the customization object
    * and if that is found, will assign all iterable values from that
-   * type into the new type, allowing default behaviour to be configured.
+   * type into the new type, allowing default behavior to be configured.
    */
   public transform(customization: Customization): Customization {
     if (!customization) {
       return customization;
     }
-    const { customizationType } = customization;
-    if (!customizationType) {
+    const { inheritsFrom } = customization;
```

Comment:
Could we make it $inheritsFrom to agree with the other transforms?  I know that means that the hasDollar needs to change, but that one should really check whether that particular operation exists anyways, so that would work generally.

Reply:
I'm not sure if that's an operator like others or not. What makes you think it is an operator? To me, it looks like a referenced id, not something that performs an operation. I changed the $transform though